lib/mix/tasks/gh_pages_docs.ex

```diff
@@ -10,36 +10,82 @@ defmodule Mix.Tasks.GhPagesDocs do
   @shortdoc "Update a documentation version on GH Pages."
 
   @spec run(term) :: :ok
-  def run(_args) do
-    version = "v#{Project.config()[:version]}"
-
-    case File.read!("assets/js/versions.js") |> String.contains?(version) do
-      true ->
-        Mix.Task.run("docs")
-        File.cd!("doc")
-        Mix.shell().cmd("git stash")
-
-        Mix.shell().cmd("git checkout gh-pages")
-
-        case File.mkdir("../#{version}") do
-          result when result in [:ok, {:error, :eexist}] ->
-            Mix.shell().cmd("cp -r ./* ../#{version}")
-            File.cd!("..")
-            Mix.shell().cmd("ls")
-            Mix.shell().cmd("git add #{version}/*")
-            Mix.shell().cmd("git commit -m \"Update #{version}\"")
-            Mix.shell().cmd("rm -rf doc")
-            Mix.shell().cmd("git push origin gh-pages")
-
-          _ ->
-            Mix.raise("Cannot create the #{version} directory")
+  def run([version]) do
+    unless File.read!("assets/js/versions.js") |> String.contains?(version) do
+      update_versions_js()
+    end
+
+    version =
+      case version do
+        "latest" -> Project.config()[:version]
+        tag -> "v#{tag}"
+      end
+
+    Mix.Task.run("docs")
+    File.cd!("doc")
+    Mix.shell().cmd("git stash")
+
+    Mix.shell().cmd("git checkout gh-pages")
+
+    case File.mkdir("../#{version}") do
+      result when result in [:ok, {:error, :eexist}] ->
+        Mix.shell().cmd("cp -r ./* ../#{version}")
+        File.cd!("..")
+        Mix.shell().cmd("ls")
+        Mix.shell().cmd("git add #{version}/*")
+
+        unless File.read!("assets/js/versions.js") |> String.contains?(version) do
+          update_versions_js()
         end
 
+        update_index_html()
+
+        Mix.shell().cmd("git add assets/js/versions.js")
+        Mix.shell().cmd("git add index.html")
+        Mix.shell().cmd("git commit -m \"Add content for #{version}\"")
+        Mix.shell().cmd("rm -rf doc")
+        Mix.shell().cmd("git show ")
+        Mix.shell().cmd("git push origin gh-pages")
+
       _ ->
-        Mix.raise("""
-        The assets/js/versions.js file is out of date.
-        Please add the newest version to the list.
-        """)
+        Mix.raise("Cannot create the #{version} directory")
     end
   end
+
+  defp update_versions_js() do
+    current = Project.config()[:version]
+    {output, 0} = System.cmd("git", ["tag"], [])
+
+    versions =
+      output
+      |> String.split("\n")
+      |> Enum.drop(7)
```

Comment:
I'm afraid this is a bit error-prone. If for some reasons a tag which today has number 4 is removed, then index 7 contains some other tag, which we probably don't want to remove. Is it possible to explicitly say which tags we are skipping?